### PR TITLE
Add `pyflakes.extend-generics` setting

### LINKED ISSUE #4654: False positive for unused import when using type as string with generic class behind TYPE_CHECKING flag
Hello, this happens when using django models with `django-types` and `ForeignKey["SomeModel"]`, but here's minimal example without django:

```python
from typing import TYPE_CHECKING, Generic, TypeVar


if TYPE_CHECKING:
    from pathlib import Path

T = TypeVar("T")


class ForeignKey(Generic[T]):
    ...


class Foo:
    var = ForeignKey["Path"]()
```

```sh
> ruff --version
ruff 0.0.270
> ruff --isolated /tmp/test.py
/tmp/test.py:5:25: F401 [*] `pathlib.Path` imported but unused
Found 1 error.
```

### PR DESCRIPTION
**Pull request by JonathanPlasse**
- Close #4654
- [ ] Improve documentation

## Summary

This setting has been added to fix a false positive where an object should be considered an annotated subscript (c.f. #4654).

## Test Plan

A fixture has been added that tests with and without the new setting.

### CODE REVIEW

File: crates/ruff_python_semantic/src/analyze/typing.rs [hunk 1]
```diff
@@ -28,13 +28,23 @@ pub fn match_annotated_subscript<'a>(
...
     if !matches!(expr, Expr::Name(_) | Expr::Attribute(_)) {
         return None;
     }
 
     model.resolve_call_path(expr).and_then(|call_path| {
-        if SUBSCRIPTS.contains(&call_path.as_slice()) {
+        let extend_annotated_subscripts: Vec<CallPath> = extend_annotated_subscripts
+            .iter()
+            .map(|target| from_qualified_name(target))
+            .collect();
```
Comment (JonathanPlasse) [inline crates/ruff_python_semantic/src/analyze/typing.rs:41]: Should this be computed in the setting initialization?
I copied `extend_immutable_calls` implementation.

Comment (MichaReiser) [inline crates/ruff_python_semantic/src/analyze/typing.rs:41]: Do we need the `collect` call here? It seems that we only iterate once.

Comment (JonathanPlasse) [reply crates/ruff_python_semantic/src/analyze/typing.rs]: Changed it.

**Comment by charliermarsh** [comment]
@JonathanPlasse - I decided to rename this, since `annotated-subscripts` is more of an internal concept.

File: crates/ruff/src/rules/pyflakes/snapshots/ruff__rules__pyflakes__tests__F401_F401_15.py.snap [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ℹ Suggested fix
+2 2 | from django.db.models import ForeignKey
+3 3 | 
+4 4 | if TYPE_CHECKING:
+5   |-    from pathlib import Path
+  5 |+    pass
+6 6 | 
+7 7 | 
+8 8 | class Foo:
```
[the remark this answers is not in the document]
Comment (last-partizan) [reply crates/ruff/src/rules/pyflakes/snapshots/ruff__rules__pyflakes__tests__F401_F401_15.py.snap:20]: Oh, great, thank you!

**Pull request merged**